generic-msp430-gcc.cmake

```diff
@@ -0,0 +1,76 @@
+#toolchain cmake file for msp430-gcc/avrdude toolchain
+
+cmake_minimum_required(VERSION 3.20)
+# 3.17 - for REQUIRED on find_program
+# 3.20 - for cmake_path
+
+# needs the following variables:
+# MSP430_MCU : mcu type 
+# MSP430_MCU_FREQ : clock frequency (defines F_CPU)
+
+
+#generic avr flags
+
+set(MSP430_CFLAGS CACHE STRING "MSP430 compilation flags")
+set(MSP430_LFLAGS "-Wl,--relax,--gc-sections" CACHE STRING "MSP430 link flags")
+
+#find toolchain programs
+set(MSP430_PREFIX "msp430-elf-")
+list(APPEND CMAKE_PREFIX_PATH "$ENV{MSP430_GCC_PATH}")
+find_program(MSP430-GCC ${MSP430_PREFIX}gcc REQUIRED)
+find_program(MSP430-GXX ${MSP430_PREFIX}g++ REQUIRED)
+find_program(MSP430-OBJCOPY ${MSP430_PREFIX}objcopy REQUIRED)
+find_program(MSP430-SIZE ${MSP430_PREFIX}size REQUIRED)
+find_program(MSP430-OBJDUMP ${MSP430_PREFIX}objdump REQUIRED)
+find_program(MSPDEBUG mspdebug)
+
+if(NOT MSP430_GCC_PATH)
+    cmake_path(SET MSP430_GCC_PATH NORMALIZE "${MSP430-GCC}/../..")
+endif(NOT MSP430_GCC_PATH)
+message(STATUS "Using directory ${MSP430_GCC_PATH} for msp430-gcc")
+
+
+#define toolchain
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_C_COMPILER ${MSP430-GCC})
+set(CMAKE_CPP_COMPILER ${MSP430-GXX})
+
+#Release by default
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
```

Comment:
Mixing tabs and spaces. Indentation is out of whack below in several places, too.

Reply:
I'll see if VSCode will fix up the formatting.